### PR TITLE
Rename several IM message with Message postfix

### PR DESCRIPTION
**Pull request by yunhanw-google**
#### Problem
Update SubscribeRequest/Response, ReadRequest, ReportData, StatusResponse， WriteRequest/Response with Message postfix

#### Change overview
See above

#### Testing
The existing presubmit test covers.

### CODE REVIEW

**Review by mrjerryjohns — APPROVED** [review]

**Comment by andy31415** [comment]
fast track: rename only with approval from domain owner

File: src/app/MessageDef/ReadRequestMessage.cpp [hunk 1]
```diff
@@ -180,7 +180,11 @@ CHIP_ERROR ReadRequest::Parser::GetAttributePathList(AttributePathList::Parser *
     return err;
 }
 
+<<<<<<< HEAD:src/app/MessageDef/ReadRequest.cpp
 CHIP_ERROR ReadRequest::Parser::GetEventPaths(EventPaths::Parser * const apEventPaths) const
+=======
+CHIP_ERROR ReadRequestMessage::Parser::GetEventPathList(EventPathList::Parser * const apEventPathList) const
+>>>>>>> 3d86bbb3d (Rename several IM message with Message postfix):src/app/MessageDef/ReadRequestMessage.cpp
```
Comment (bzbarsky-apple) [inline src/app/MessageDef/ReadRequestMessage.cpp]: This is not going to go well.

File: src/app/ReadClient.h [hunk 2]
```diff
@@ -178,19 +179,19 @@ class ReadClient : public Messaging::ExchangeDelegate
...
     {
-        Uninitialized = 0,         ///< The client has not been initialized
-        Initialized,               ///< The client has been initialized and is ready for a SendReadRequest
-        AwaitingInitialReport,     ///< The client is waiting for initial report
-        AwaitingSubscribeResponse, ///< The client is waiting for subscribe response
-        SubscriptionActive,        ///< The client is maintaining subscription
+        Uninitialized = 0,                ///< The client has not been initialized
+        Initialized,                      ///< The client has been initialized and is ready for a SendReadRequestMessage
+        AwaitingInitialReport,            ///< The client is waiting for initial report
+        AwaitingSubscribeResponseMessage, ///< The client is waiting for subscribe response
```
Comment (bzbarsky-apple) [inline src/app/ReadClient.h:193]: I'm not sure this use should be renamed....

File: src/app/InteractionModelEngine.h [hunk 3]
```diff
@@ -95,22 +95,22 @@ class InteractionModelEngine : public Messaging::ExchangeDelegate, public Comman
...
     /**
-     *  Creates a new read client and send ReadRequest message to the node using the read client,
+     *  Creates a new read client and send ReadRequestMessage message to the node using the read client,
      *  shutdown if fail to send it out
      *
      *  @retval #CHIP_ERROR_NO_MEMORY If there is no ReadClient available
      *  @retval #CHIP_NO_ERROR On success.
      */
-    CHIP_ERROR SendReadRequest(ReadPrepareParams & aReadPrepareParams, ReadClient::Callback * aCallback);
+    CHIP_ERROR SendReadRequestMessage(ReadPrepareParams & aReadPrepareParams, ReadClient::Callback * aCallback);
```
Comment (bzbarsky-apple) [inline src/app/InteractionModelEngine.h:104]: I'm not sure the renaming is the right thing here.  The idea here is to send a read request action, not just a message...

File: src/app/InteractionModelEngine.h [hunk 4]
```diff
@@ -95,22 +95,22 @@ class InteractionModelEngine : public Messaging::ExchangeDelegate, public Comman
...
     /**
-     *  Creates a new read client and sends SubscribeRequest message to the node using the read client.
+     *  Creates a new read client and sends SubscribeRequestMessage message to the node using the read client.
      *  Shuts down on transmission failure.
      *
      *  @retval #CHIP_ERROR_NO_MEMORY If there is no ReadClient available
      *  @retval #CHIP_NO_ERROR On success.
      */
-    CHIP_ERROR SendSubscribeRequest(ReadPrepareParams & aReadPrepareParams, ReadClient::Callback * aCallback);
+    CHIP_ERROR SendSubscribeRequestMessage(ReadPrepareParams & aReadPrepareParams, ReadClient::Callback * aCallback);
```
Comment (bzbarsky-apple) [inline src/app/InteractionModelEngine.h:113]: Again, I don't know that we want the renaming here.

File: src/app/ReadClient.h [hunk 5]
```diff
@@ -124,7 +124,7 @@ class ReadClient : public Messaging::ExchangeDelegate
          * This function will:
          *      - Always be called exactly *once* for a given WriteClient instance.
          *      - Be called even in error circumstances.
-         *      - Only be called after a successful call to SendWriteRequest as been made.
+         *      - Only be called after a successful call to SendWriteRequestMessage as been made.
```
Comment (bzbarsky-apple) [inline src/app/ReadClient.h:127]: I don't understand what `SendWriteRequestMessage` has to do with `ReadClient`.

File: src/app/ReadClient.h [hunk 6]
```diff
@@ -144,30 +144,31 @@ class ReadClient : public Messaging::ExchangeDelegate
...
      *  Send a Read Request.  There can be one Read Request outstanding on a given ReadClient.
-     *  If SendReadRequest returns success, no more Read Requests can be sent on this ReadClient
+     *  If SendReadRequestMessage returns success, no more Read Requests can be sent on this ReadClient
      *  until the corresponding InteractionModelDelegate::ReadDone call happens with guarantee.
      *
      *  @retval #others fail to send read request
      *  @retval #CHIP_NO_ERROR On success.
      */
-    CHIP_ERROR SendReadRequest(ReadPrepareParams & aReadPrepareParams);
+    CHIP_ERROR SendReadRequestMessage(ReadPrepareParams & aReadPrepareParams);
```
Comment (bzbarsky-apple) [inline src/app/ReadClient.h:159]: Why are we renaming this?  The important thing is that we are sending a read request action.  Whether it's one message or not is implementation detail here.

File: src/app/ReadClient.h [hunk 7]
```diff
@@ -144,30 +144,31 @@ class ReadClient : public Messaging::ExchangeDelegate
...
-     *  If SendSubscribeRequest returns success, no more subscribe Requests can be sent on this ReadClient
+     *  If SendSubscribeRequestMessage returns success, no more subscribe Requests can be sent on this ReadClient
      *  until the corresponding InteractionModelDelegate::ReadDone call happens with guarantee.
      *
      *  @retval #others fail to send subscribe request
      *  @retval #CHIP_NO_ERROR On success.
      */
-    CHIP_ERROR SendSubscribeRequest(ReadPrepareParams & aSubscribePrepareParams);
-    CHIP_ERROR OnUnsolicitedReportData(Messaging::ExchangeContext * apExchangeContext, System::PacketBufferHandle && aPayload);
+    CHIP_ERROR SendSubscribeRequestMessage(ReadPrepareParams & aSubscribePrepareParams);
```
Comment (bzbarsky-apple) [inline src/app/ReadClient.h:169]: Again, not sure this should have the "Message" suffix.

File: src/app/WriteClient.h [hunk 8]
```diff
@@ -47,7 +47,7 @@ class InteractionModelEngine;
  *  @brief The read client represents the initiator side of a Write Interaction, and is responsible
  *  for generating one Write Request for a particular set of attributes, and handling the Write response.
  *  Consumer can allocate one write client, then call PrepareAttribute, insert attribute value, followed by FinishAttribute for
- * every attribute it wants to insert in write request, then call SendWriteRequest
+ * every attribute it wants to insert in write request, then call SendWriteRequestMessage
```
Comment (bzbarsky-apple) [inline src/app/WriteClient.h:50]: But with chunking it would not be a single write request message anyway....

File: src/app/WriteClient.h [hunk 9]
```diff
@@ -136,15 +136,15 @@ class WriteClient : public Messaging::ExchangeDelegate
...
      *  handle calling Shutdown on itself once it decides it's done with waiting
      *  for a response (i.e. times out or gets a response). Client can specify
      *  the maximum time to wait for response (in milliseconds) via timeout parameter.
      *  Default timeout value will be used otherwise.
-     *  If SendWriteRequest is never called, or the call fails, the API
+     *  If SendWriteRequestMessage is never called, or the call fails, the API
      *  consumer is responsible for calling Shutdown on the WriteClient.
      */
-    CHIP_ERROR SendWriteRequest(SessionHandle session, System::Clock::Timeout timeout);
+    CHIP_ERROR SendWriteRequestMessage(SessionHandle session, System::Clock::Timeout timeout);
```
Comment (bzbarsky-apple) [inline src/app/WriteClient.h:147]: No, this might be sending multiple messages.  This should stay `SendWriteRequest`.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
I didn't check every line touched carefully, but for a number of those I did check adding "Message" does not make sense to me.

Updating the parser bits and the bits that deal with actual messages makes sense.  Changing higher-level APIs that really are operating on the level of "write request", not messages, does not make sense.

File: src/app/MessageDef/ReadRequestMessage.cpp (hunk 1 above)
Comment (yunhanw-google) [reply src/app/MessageDef/ReadRequestMessage.cpp]: Merge conflict...Thanks for catching it up. Fixed.

File: src/app/ReadClient.h (hunk 2 above)
Comment (yunhanw-google) [reply src/app/ReadClient.h:193]: sure, rename it back.

File: src/app/InteractionModelEngine.h (hunk 3 above)
Comment (yunhanw-google) [reply src/app/InteractionModelEngine.h:104]: sure, rename it back.

File: src/app/ReadClient.h (hunk 5 above)
Comment (yunhanw-google) [reply src/app/ReadClient.h:127]: sure, rename it back.

File: src/app/ReadClient.h (hunk 7 above)
Comment (yunhanw-google) [reply src/app/ReadClient.h:169]: sure, rename it back.

File: src/app/WriteClient.h (hunk 9 above)
Comment (yunhanw-google) [reply src/app/WriteClient.h:147]: sure, rename it back.

**Comment by bzbarsky-apple** [comment]
To be clear: Please go through all the changes and undo the ones that don't make sense.  I mentioned some of them in my review comments, but I did not do a complete careful read.  I'd really prefer to only have to do that once....

**Comment by yunhanw-google** [comment]
@bzbarsky-apple agree on your suggestion, just updated, thanks

File: src/app/WriteClient.h (hunk 8 above)
Comment (yunhanw-google) [reply src/app/WriteClient.h]: Rename it back, thanks

File: src/app/ReadClient.h (hunk 6 above)
Comment (yunhanw-google) [reply src/app/ReadClient.h]: Rename it back, thanks

File: src/app/InteractionModelEngine.h (hunk 4 above)
Comment (yunhanw-google) [reply src/app/InteractionModelEngine.h]: Rename it back, thanks

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**